### PR TITLE
Generic PostFeed component

### PR DESCRIPTION
**Pull request by snowkat**
This PR depends on #14, mainly because I gave up dealing with merge conflicts on the post components.

This creates a PostFeed component, responsible for the actual feed that appears as the timeline, or will appear on profile pages.

For the most part it's working, but two caveats:

- [x] Without a direct way to reset the feed, we're back to the broken state where posting while on `/feed` doesn't update the home timeline. Current plan is to use the FeedContext to create some kind of "reloader" component, but not sure if this is the best idea.

- [ ] Back button. Maybe I'm overthinking it, but is there a reliable way to determine what the "first" page is in a stateless way? My first thought is Load last 25 -> Load 25 before that -> If 0 posts, we're on the first page. but that's really noisy, and would lead to weird situations where the new first page might show an oddly small number of posts.

### CODE REVIEW

**Comment by vivlim** [comment]
no worries about the dependency on #14 👍

regarding the refresh after post issue, will we also need to solve that for adding comments to posts? that seems kinda similar.

I'll spend some time reviewing later.

**Comment by vivlim** [comment]
maybe this is a silly hack but... could PostFeedProps also contain the current timestamp? i think it kinda makes some sense because one of the inputs deciding what the feed will consist of *is* when you poll it. then when adding a post, the timestamp could be bumped which I think(?) would trigger a refresh

**Comment by snowkat** [comment]
> maybe this is a silly hack but... could PostFeedProps also contain the current timestamp? i think it kinda makes some sense because one of the inputs deciding what the feed will consist of _is_ when you poll it. then when adding a post, the timestamp could be bumped which I think(?) would trigger a refresh

ayyy that did it :sunglasses:

**Comment by snowkat** [comment]
> regarding the refresh after post issue, will we also need to solve that for adding comments to posts? that seems kinda similar.

Almost certainly, but a similar method for refreshing should do the trick.

**Comment by vivlim** [comment]
what is the back button issue? if it's #15, that's an existing issue which you don't need to block merging on

File: src/components/post/feed/index.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        },
+    };
+
+    return (
+        <FeedContext.Provider value={feedContext}>
+            <div>
+                <ErrorBoundary fallback={<div>Failed to load posts.</div>}>
+                    <For each={postList()}>
+                        {(status, index) => (
+                            <Post status={status} fetchShareParent={false} />
```
Comment (vivlim) [inline src/components/post/feed/index.tsx:105]: huh, I thought this was true at some point, but it's false on the left side too.

i guess that could be why I haven't seen shares in the feed in a bit

**Comment by snowkat** [comment]
> what is the back button issue? if it's https://github.com/vivlim/pillbug/issues/15, that's an existing issue which you don't need to block merging on

It's a feature that already doesn't exist, but if you're a few "pages" into your feed and assuming previous pages aren't in your browser history, I don't know of a good way to statelessly go "back" a page. Trying to fake it by using `since_id: searchParams.after` might work, but page 0 could end up being one or two posts.

Still maybe not worth blocking this on, especially since it's just... not adding a feature that didn't exist yet.

**Review by vivlim — APPROVED** [review]

**Comment by vivlim** [comment]
i see, yeah I'm inclined to proceed without it for now.

a thought is maybe having a signal that stores the first post id inside of the async fetch function, but I think it'd be better to fix browser navigation first so that it doesn't jump to the top. a previous page button *could* just navigate backward in browser history, since generally the only way to get to a specific page is to click the next button one or more times .

**Comment by vivlim** [comment]
i guess there are some edge cases like, saving a bookmark which happens to contain a query parameter, but... meh

**Pull request merged**